### PR TITLE
Add code cell symbols to nb outline + debt fixes

### PR DESCRIPTION
**Pull request by Yoyokrazy**
re #206308

Includes:
- code cell symbols as children of code cells in the notebook outline
- filtering logic moved out of the outlineProvider into getChildren of notebookCellOutline
- stickyScroll adjusted for new level 8 outlineEntry (symbol entries)

### CODE REVIEW

**Comment by Yoyokrazy** [comment]
todo:
- [ ] need to fix some notebook symbol tests

File: src/vs/workbench/contrib/notebook/browser/contrib/outline/notebookOutline.ts [hunk 1]
```diff
@@ -602,25 +660,26 @@ registerAction2(class ToggleCodeCellEntries extends Action2 {
...
 			}
 		});
 	}
 
 	run(accessor: ServicesAccessor, ...args: any[]) {
 		const configurationService = accessor.get(IConfigurationService);
-		const showNonHeaderMarkdownCells = configurationService.getValue<boolean>('notebook.outline.showNonHeaderMarkdownCells');
-		configurationService.updateValue('notebook.outline.showNonHeaderMarkdownCells', !showNonHeaderMarkdownCells);
+		const showCodeCellSymbols = configurationService.getValue<boolean>('notebook.outline.showCodeCellSymbols');
+		configurationService.updateValue('notebook.outline.showCodeCellSymbols', !showCodeCellSymbols);
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/contrib/outline/notebookOutline.ts:683]: Maybe we can create consts for all outline symbols settings other than using string literals.

Comment (Yoyokrazy) [reply src/vs/workbench/contrib/notebook/browser/contrib/outline/notebookOutline.ts]: resolved in latest commit. We now have NotebookSetting.xyz for these settings

**Comment by Yoyokrazy** [comment]
updates:
- settings defined in `notebookCommon` for outline showCodeCells, showCodeCellSymbols, and showMarkdownHeadersOnly.
- settings defined in `notebookCommon` for breadcrumbs showCodeCells
- filter logic moved to `notebookOutline` getChildren fn
- symbols from code cells are now OutlineEntry level 8+ (keeps them as children of code cells in outlinePane)
- outlineEntryFactory now takes the OutlineTarget (helps control whether a code cell entry is pushed. needed in outlinePane, not in quickPick)

**Review by rebornix — APPROVED** [review]

**Pull request merged**